LICENSE.AGPL

```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```

Comment:
Yes, it would be nice if the code can run different protocols without modifying the code, but that's an orthogonal (technical) issue we should discuss separately.

If the problem you see there is that all the protobuf protocols currently supported are "wired together" via nested interlocking message/struct definitions using only small-integer codes to distinguish them, we already discussed that.  To reiterate what my memory of the plan was, for simplicity and efficiency we would just use the direct protobuf nested-message/struct approach for "native" or "intended-to-be-stable" protocols, but then (eventually) have an explicit "extension message/protocol" option that uses ASCII strings instead of small-ints to identify arbitrary extension protocols, which would ideally be pluggable without modifying the existing code.

But to be clear, even then, this will not mean that "only DEDIS" can add new "native/stable" protocols to the protobuf definitions.  It will simply mean that anyone who wants to add new native/stable protocols will need to modify the code (which anyone can do), whereas adding "experimental" protocols via the ASCI-string-identified extension mechanism will be a bit easier (and encouraged for experimental protocols anyway).  Does this make sense?  Or am I barking up the wrong tree (i.e., discussing the wrong problem)?